lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,49 +1,90 @@
-        require("ace/ext/jquery")
-        katex = null
-
-        initKaTeX = (onLoaded) ->
-            # Adding KaTeX CSS
-            cssKatexPath = require.toUrl("../katex/katex.min.css")
-            linkKatex = $("<link>").attr("rel", "stylesheet").attr("href", cssKatexPath)
-            $("head").append(linkKatex)
-
-            # Adding CSS for demo formula
-            cssDemoPath = require.toUrl("./katex-demo.css")
-            linkDemo = $("<link>").attr("rel", "stylesheet").attr("href", cssDemoPath)
-            $("head").append(linkDemo)
-
-            # Adding DOM element to place formula into
-            span = $("<span>").attr("id", "formula")
-            $("body").append(span)
-
-            require(["ace/ext/katex/katex"], (katexInner) ->
-                katex = katexInner
-                onLoaded()
-                return
-            )
-            return
-
-        onLoaded = ->
-            selectedText = editor.getSelectedText()
-            try
-                katex.render(selectedText, $("#formula")[0])
-            catch e
-                $("#formula").text(e.message)
-            return
-
-        callback = (editor) ->
-            unless katex?
-                initKaTeX(onLoaded)
-                return
-            onLoaded()
-
-        editor.commands.addCommand({
-            name: "previewLaTeXFormula",
-            bindKey: {win: "Alt-p", mac: "Alt-p"},
-            exec: callback
-        })
+  exports.setupPreviewer = (editor, popoverHandler) ->
+    katex = null
+    popoverHandler = popoverHandler ? {
+      show: (jqPopoverContainer, options) ->
+        cursorPosition = $("textarea.ace_text-input").position()
+        cursorPosition.top += 24
+        cursorPosition.top += "px"
+        cursorPosition.left += "px"
+        jqPopoverContainer.css(cursorPosition)
+        jqPopoverContainer.popover(options)
+        jqPopoverContainer.popover("show")
         return
+
+      hide: (jqPopoverContainer) ->
+        jqPopoverContainer.popover("destroy")
+
+      isVisible: (popoverElement) ->
+        popoverElement.children(".popover").is(":visible")
+    }
+
+    initKaTeX = (onLoaded) ->
+      # Adding CSS for demo formula
+      cssDemoPath = require.toUrl("./katex-demo.css")
+      linkDemo = $("<link>").attr(
+        rel: "stylesheet"
+        href: cssDemoPath
+      )
+      $("head").append(linkDemo)
+
+      # Adding DOM element to place formula into
+      a = $("<span>").attr(
+        id: "formula"
+      )
+      $("body").append(a)
+
+      require(["ace/ext/katex"], (katexInner) ->
+        katex = katexInner
+        onLoaded()
+        return
+      )
+      return
+
+    callbackHidePopover = ->
+      popoverHandler.hide($("#formula"))
+      editor.off("changeSelection", callbackHidePopover)
+      editor.session.off("changeScrollTop", callbackHidePopover)
+      editor.session.off("changeScrollLeft", callbackHidePopover)
+      return
+
+    onLoaded = ->
+      options = {
```

Comment:
I don't think I can do it in the way that makes sense, because `options` contain `content` field, which actually determine, what is in the popover.